cpp/src/io/orc/writer_impl.cu

```diff
@@ -79,11 +80,6 @@ struct row_group_index_info {
 };
 
 namespace {
-/**
- * @brief Helper for pinned host memory
- */
-template <typename T>
-using pinned_buffer = std::unique_ptr<T, decltype(&cudaFreeHost)>;
```

Comment:
This looks good, with one minor concern. For places that already used `host_vector`, there's no change. But, for this use case, we are introducing initialization into the memory that does not need to be initialized (and previously wasn't). If you don't mind, please run ORC or Parquet writer benchmarks.

Reply:
Sorry I didn't see where is the initialization that you mentioned? The new `pinned_host_vector` uses allocator that also doesn't initialize the internal buffer:
```
__host__ inline pointer allocate(size_type cnt, const_pointer /*hint*/ = 0)
  {
    if (cnt > this->max_size()) { throw std::bad_alloc(); }  // end if

    pointer result(0);
    CUDF_CUDA_TRY(cudaMallocHost(reinterpret_cast<void**>(&result), cnt * sizeof(value_type)));
    return result;
  }
```

Reply:
Please see benchmarks here: https://github.com/rapidsai/cudf/pull/13206#issuecomment-1520782754

Reply:
I thought that vector has its own initialization outside of the allocator. Either way, it does not seem to impact the overall performance.